lib/logstash/outputs/elasticsearch/shared_configs.rb

```diff
@@ -0,0 +1,160 @@
+require 'forwardable' # Needed for logstash core SafeURI. We need to patch this in core: https://github.com/elastic/logstash/pull/5978
+
+module LogStash; module Outputs; class ElasticSearch
+  module SharedConfigs
```

Comment:
yes, these are shared, but _why_ are they shared?

I think `ConnectionConfigs` may be a better name because almost all of them seem to have to do with establishing and maintaining connections to Elasticsearch. There are a couple around retries that don't seem to quite fit there, so maybe `APIConfigs`?

Reply:
Agreed naming is a bit hard here and SharedConfigs is certainly not the best. +1 APIConfigs.